src/controller/CHIPDeviceController.cpp

```diff
@@ -571,13 +571,16 @@ CHIP_ERROR DeviceCommissioner::PairDevice(NodeId remoteDeviceId, const char * se
         return CHIP_ERROR_INCORRECT_STATE;
     }
     ReturnErrorOnFailure(mDefaultCommissioner->SetCommissioningParameters(params));
-    return mSetUpCodePairer.PairDevice(remoteDeviceId, setUpCode, SetupCodePairerBehaviour::kCommission);
+    return PairDevice(remoteDeviceId, setUpCode);
 }
 
 CHIP_ERROR DeviceCommissioner::PairDevice(NodeId remoteDeviceId, const char * setUpCode)
 {
     MATTER_TRACE_EVENT_SCOPE("PairDevice", "DeviceCommissioner");
-    return mSetUpCodePairer.PairDevice(remoteDeviceId, setUpCode, SetupCodePairerBehaviour::kCommission);
+    ReturnErrorOnFailure(mSetUpCodePairer.PairDevice(remoteDeviceId, setUpCode, SetupCodePairerBehaviour::kCommission));
+
+    return mSystemState->SystemLayer()->StartTimer(chip::System::Clock::Milliseconds32(kSessionEstablishmentTimeout),
+                                                   mSetUpCodePairer.OnSessionEstablishmentTimeoutCallback, &mSetUpCodePairer);
```

Comment:
1. Why is this timer started here, and not inside the setup code pairer?
2. Why is this timer happening only in the `kCommission` cases. and not in all the uses of setup code pairer that just establish PASE sessions?
3. What cancels this timer?  If we complete one session establishement, what keeps it from firing in the middle of a second one, if the second one just does a PASE session, not commissioning?
4. Is it really the intent for this timer to interrupt an ongoing PASE connection attempt based on an advertisement we discovered?  Does it actually interrupt that, or leave it to keep establishing but notify that session establishment failed?